ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,866 @@
+use constant DOMAINNAME          => "/system/network/domainname";
+use constant REPOS_DIR           => "/etc/yum.repos.d";
+use constant REPOS_TREE          => "/software/repositories";
+use constant PKGS_TREE           => "/software/packages";
+use constant GROUPS_TREE         => "/software/groups/names";
+use constant CMP_TREE            => "/software/components/spma";
+use constant YUM_PACKAGE_LIST    => "/etc/yum/pluginconf.d/versionlock.list";
+use constant YUM_CONF_FILE       => "/etc/yum.conf";
+use constant RPM_QUERY_INSTALLED => qw(rpm -qa --nosignature --nodigest --qf %{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n);
+use constant RPM_QUERY_INSTALLED_NAMES => qw(rpm -qa --nosignature --nodigest --qf %{EPOCH}:%{NAME}\n);
+use constant REPO_AVAIL_PKGS     => qw(repoquery -C --show-duplicates --all --qf %{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH});
+use constant YUM_PLUGIN_OPTS     => "--disableplugin=\* --enableplugin=fastestmirror --enableplugin=versionlock --enableplugin=priorities";
+use constant YUM_TEST_CHROOT     => qw(/tmp/spma_yum_testroot);
+use constant SPMA                => "/software/components/spma";
+use constant SPMAPROXY           => "/software/components/spma/proxy";
+
+our $NoActionSupported = 1;
+
+# If user packages are not allowed, removes any repositories present
+# in the system that are not listed in $allowed_repos.
+sub cleanup_old_repos
+{
+    my ( $self, $repo_dir, $allowed_repos, $allow_user_pkgs ) = @_;
+
+    return 1 if $allow_user_pkgs;
+
+    my $dir;
+    if ( !opendir( $dir, $repo_dir ) ) {
+        $self->error("Unable to read repositories in $repo_dir");
+        return 0;
+    }
+
+    my $current = Set::Scalar->new( map( m{(.*)\.repo$}, readdir($dir) ) );
+
+    closedir($dir);
+    my $allowed = Set::Scalar->new( map( $_->{name}, @$allowed_repos ) );
+    my $rm = $current - $allowed;
+    foreach my $i (@$rm) {
+        # We use $f here to make Devel::Cover happy
+	if ($i =~ m/^[\w-]+$/) {
+	    my $f = "$repo_dir/$i.repo";
+	    $self->verbose("Unlinking outdated repository $f");
+	    if ( !unlink($f) ) {
+		$self->error("Unable to remove outdated repository $i: $!");
+		return 0;
+	    }
+	} else {
+	    return 0;
+	}
+    }
+    return 1;
+}
+
+=pod
+
+=head2 C<execute_yum_command>
+
+Executes C<$command> for reason C<$why>. Optionally with standard
+input C<$stdin>.  The command may be executed even under --noaction if
+C<$keeps_state> has a true value.
+
+If the command is executed, this method returns its standard output
+upon success or C<undef> in case of error.  If the command is not
+executed the method always returns a true value (usually 1, but don't
+rely on this!).
+
+Yum-based commands require annoying error handling: they may exit 0
+even upon errors.  In those cases, we have to detect errors by looking
+at stderr.  This method just encapsulates all that logic, keeping the
+callers clean.
+
+=cut
+
+my $os_major = "";
```

Comment:
i'm sorry, but i can't let you do this :smile:. define them where needed, unless they are truly intended to be used as class variables in instead of convenient way to redefine them.